src/main/java/bio/terra/workspace/app/configuration/spring/TraceInterceptorConfig.java

```diff
@@ -0,0 +1,113 @@
+package bio.terra.workspace.app.configuration.spring;
+
+import static org.apache.commons.lang3.ObjectUtils.firstNonNull;
+
+import bio.terra.workspace.app.configuration.external.TracingConfiguration;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceConfiguration;
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceExporter;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.samplers.Samplers;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.hashids.Hashids;
+import org.slf4j.MDC;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+@ConditionalOnProperty("workspace.tracing.enabled")
+public class TraceInterceptorConfig implements WebMvcConfigurer {
+
+  public static final String MDC_REQUEST_ID_HEADER = "X-Request-ID";
+  public static final String MDC_REQUEST_ID_KEY = "requestId";
+  public static final String MDC_CORRELATION_ID_HEADER = "X-Correlation-ID";
+
+  private final Hashids hashids = new Hashids("requestIdSalt", 8);
+
+  @Autowired
+  public TraceInterceptorConfig(
+      TracingConfiguration tracingConfiguration, Environment environment) {
+    try {
+      StackdriverTraceExporter.createAndRegister(
+          StackdriverTraceConfiguration.builder()
+              .setProjectId(tracingConfiguration.getProjectId())
+              .setCredentials(
+                  ServiceAccountCredentials.fromStream(
+                      new FileInputStream(tracingConfiguration.getSaPath())))
+              .setFixedAttributes(
+                  Map.of(
+                      "/component",
+                      AttributeValue.stringAttributeValue(
+                          environment.getRequiredProperty("spring.application.name"))))
+              .build());
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    TraceConfig globalTraceConfig = Tracing.getTraceConfig();
+    globalTraceConfig.updateActiveTraceParams(
+        globalTraceConfig.getActiveTraceParams().toBuilder()
+            .setSampler(Samplers.probabilitySampler(tracingConfiguration.getProbability()))
+            .build());
```

Comment:
optional: This is generic StackdriverTraceExporter configuration set up. It's not specific to the TraceInterceptor any more than any other place we start use traces. Consider giving Stackdriver its own configuration class that just does this setup and keep this class as only the work needed to adding spans/attributes to each incoming request.
That's a natural division as OpenCensus encourages us to plugin different exporters behind the same tracing api.

Reply:
I think I did what you were describing here.

Reply:
I meant split this into a separate class, but up to you.

Reply:
Ah, that's what I did if you look in the files tab, sorry

Reply:
oops, sorry!